beacon-chain/rpc/eth/node/handlers.go

```diff
@@ -32,3 +56,363 @@ func (s *Server) GetSyncStatus(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "node.GetHealth")
+	defer span.End()
+
+	ok, rawSyncingStatus, syncingStatus := shared.UintFromQuery(w, r, "syncing_status")
+	// lint:ignore uintcast -- custom syncing status being outside of range is harmless
+	intSyncingStatus := int(syncingStatus)
+	if !ok || (rawSyncingStatus != "" && http.StatusText(intSyncingStatus) == "") {
+		http2.HandleError(w, "syncing_status is not a valid HTTP status code", http.StatusBadRequest)
+		return
+	}
+
+	if s.SyncChecker.Synced() {
+		return
+	}
+	if s.SyncChecker.Syncing() || s.SyncChecker.Initialized() {
+		if rawSyncingStatus != "" {
+			w.WriteHeader(intSyncingStatus)
+		} else {
+			w.WriteHeader(http.StatusPartialContent)
+		}
+		return
+	}
+	w.WriteHeader(http.StatusServiceUnavailable)
+}
+
+func handleEmptyFilters(states []string, directions []string) (emptyState, emptyDirection bool) {
+	emptyState = true
+	for _, stateFilter := range states {
+		normalized := strings.ToUpper(stateFilter)
+		filterValid := normalized == stateConnecting || normalized == stateConnected ||
+			normalized == stateDisconnecting || normalized == stateDisconnected
+		if filterValid {
+			emptyState = false
+			break
+		}
+	}
+
+	emptyDirection = true
+	for _, directionFilter := range directions {
+		normalized := strings.ToUpper(directionFilter)
+		filterValid := normalized == directionInbound || normalized == directionOutbound
+		if filterValid {
+			emptyDirection = false
+			break
+		}
+	}
+
+	return emptyState, emptyDirection
+}
+
+func peerInfo(peerStatus *peers.Status, id peer.ID) (*Peer, error) {
+	enr, err := peerStatus.ENR(id)
+	if err != nil {
+		if errors.Is(err, peerdata.ErrPeerUnknown) {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "could not obtain ENR")
+	}
+	var serializedEnr string
+	if enr != nil {
+		serializedEnr, err = p2p.SerializeENR(enr)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not serialize ENR")
+		}
+	}
+	address, err := peerStatus.Address(id)
+	if err != nil {
+		if errors.Is(err, peerdata.ErrPeerUnknown) {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "could not obtain address")
+	}
+	connectionState, err := peerStatus.ConnectionState(id)
+	if err != nil {
+		if errors.Is(err, peerdata.ErrPeerUnknown) {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "could not obtain connection state")
+	}
+	direction, err := peerStatus.Direction(id)
+	if err != nil {
+		if errors.Is(err, peerdata.ErrPeerUnknown) {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "could not obtain direction")
+	}
+	if eth.PeerDirection(direction) == eth.PeerDirection_UNKNOWN {
+		return nil, nil
+	}
+	p := &Peer{
+		PeerId:    id.Pretty(),
+		State:     strings.ToLower(eth.ConnectionState(connectionState).String()),
+		Direction: strings.ToLower(eth.PeerDirection(direction).String()),
+	}
+	if address != nil {
+		p.LastSeenP2PAddress = address.String()
+	}
+	if serializedEnr != "" {
+		p.Enr = "enr:" + serializedEnr
```

Comment:
a space can be added here

Reply:
I can't because it will not conform to the specification. From https://eips.ethereum.org/EIPS/eip-778:
> The textual form of a node record is the base64 encoding of its RLP representation, prefixed by `enr:`.